R/mlClassificationNeuralNetwork.R

```diff
@@ -0,0 +1,265 @@
+    
+    accuracyStore           <- numeric(options[["maxGen"]])
+    trainAccuracyStore      <- numeric(options[["maxGen"]])
+    
+    # For plotting
+    plot_x            <- numeric()
+    plot_y            <- numeric()
+    plot_type         <- character()
+    
+    startProgressbar(options[["maxGen"]], gettext("Optimizing network topology"))
+    
+    # First generation
+    population <- .neuralNetworkOptim_init(options)
+    
+    # Fit and reproduce
+    for (gen in 1:options[["maxGen"]]) {
+      
+      progressbarTick()
+      
+      fitness                <- numeric(options[["genSize"]])
+      tmp_trainAccuracyStore <- numeric(options[["genSize"]])
+      
+      for (i in 1:length(population)) {
+        
+        p <- try({
+          nfit_valid  <- neuralnet::neuralnet(formula = formula, 
+                                              data = train, 
+                                              hidden = population[[i]]$structure,
+                                              learningrate = options[["learningRate"]],
+                                              threshold = options[["threshold"]],
+                                              stepmax = options[["stepMax"]],
+                                              rep = 1,
+                                              startweights = NULL,
+                                              algorithm = options[["algorithm"]],
+                                              err.fct = "sse",# jaspResults[["errfct"]]$object,
+                                              act.fct = jaspResults[["actfct"]]$object, 
+                                              linear.output = if (options[["actfct"]] == "linear") TRUE else FALSE)
+        })
+        
+        validPredictions <- levels(train[, options[["target"]]])[max.col(predict(nfit_valid, newdata = valid))]
+        fitness[i] <- sum(diag(prop.table(table(validPredictions, valid[, options[["target"]]]))))
+        trainPredictions <- levels(train[, options[["target"]]])[max.col(predict(nfit_valid, newdata = train))]
+        tmp_trainAccuracyStore[i] <- sum(diag(prop.table(table(trainPredictions, train[, options[["target"]]]))))
+        
+        population[[i]][["fitness"]] <- fitness[i]
+        population[[i]][["age"]] <- population[[i]][["age"]] + 1
+        
+        plot_x <- c(plot_x, gen, gen)
+        plot_y <- c(plot_y, fitness[i], tmp_trainAccuracyStore[i])
+        plot_type <- c(plot_type, "Validation", "Training")   
+      }
+      
+      # Find out best performance
+      best_fit <- order(-fitness)[1]
+      structure <- population[[best_fit]]$structure # Best performing network structure
+      
+      # For plotting we store the mean accuracy of the generation
+      accuracyStore[gen] <- mean(fitness)
+      trainAccuracyStore[gen] <- mean(tmp_trainAccuracyStore)
+      
+      # Stop when maximum generations is reached
+      if (gen == options[["maxGen"]])
+        break
```

Comment:
I prefer `break` to be called as `break()`so it's clear that a function is being called. @vandenman your opinion?